inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/reorder/reorder_kernel_bfyx_to_blocked_format.cpp

```diff
@@ -0,0 +1,263 @@
+    k.EnableOutputLayout(DataLayout::b_fs_zyx_fsv32);
+    k.EnableOutputLayout(DataLayout::bs_fs_yx_bsv16_fsv16);
+    k.EnableOutputLayout(DataLayout::bs_fs_zyx_bsv16_fsv16);
+
+    k.EnableDifferentTypes();
+    k.EnableBatching();
+    k.EnableTensorOffset();
+    k.EnableTensorPitches();
+
+    return k;
+}
+
+static inline std::string GetTiledInputOrder(size_t size) {
+    std::string order_str = "";
+    switch (size) {
+    case 4:
+        order_str = "b, f + lh, y, x";
+        break;
+    case 5:
+        order_str = "b, f + lh, z, y, x";
+        break;
+    default: throw std::runtime_error("Unsupported combination\n");
+    }
+    return order_str;
+}
+
+static inline size_t GetFsvAlignment(const reorder_params& params) {
+    const auto& out = params.output;
+    int fsv_alignment = -1;
+    switch (out.GetLayout()) {
+    case DataLayout::b_fs_yx_fsv4:
+        fsv_alignment = 4;
+        break;
+    case DataLayout::b_fs_yx_fsv16:
+    case DataLayout::b_fs_zyx_fsv16:
+    case DataLayout::bs_fs_yx_bsv16_fsv16:
+    case DataLayout::bs_fs_zyx_bsv16_fsv16:
+        fsv_alignment = 16;
+        break;
+    case DataLayout::b_fs_yx_fsv32:
+    case DataLayout::fs_b_yx_fsv32:
+    case DataLayout::b_fs_zyx_fsv32:
+        fsv_alignment = 32;
+        break;
+    default:
+        throw std::runtime_error("Unsupported combination\n");
+    }
+    return fsv_alignment;
+}
+
+static inline size_t GetTileSize(const reorder_params& params) {
+    const Datatype input_type = params.inputs[0].GetDType();
+    const Datatype output_type = params.output.GetDType();
+
+    // i64 supports tile size 4
+    if ((input_type == Datatype::INT64) || (output_type == Datatype::INT64)) {
+        return MIN_TILE_SIZE;
+    }
+
+    if (params.output.GetLayout() == DataLayout::b_fs_yx_fsv4) {
+        return MIN_TILE_SIZE;
+    }
+
+    if (params.inputs[0].Feature().v < DEFAULT_TILE_SIZE) {
+        return MIN_TILE_SIZE;
+    }
+
+    return DEFAULT_TILE_SIZE;
+}
+
+static inline std::vector<size_t> GetGWS(const reorder_params& params) {
+    const auto& in = params.inputs[0];
+    const size_t tile_size = GetTileSize(params);
+    const size_t fsv_alignment = GetFsvAlignment(params);
+    std::vector<size_t> gws;
+
+    switch (params.output.GetLayout()) {
+    case DataLayout::b_fs_yx_fsv4:
+    case DataLayout::b_fs_yx_fsv16:
+    case DataLayout::b_fs_yx_fsv32:
+    case DataLayout::fs_b_yx_fsv32:
+    case DataLayout::bs_fs_yx_bsv16_fsv16:
+        gws = { CeilDiv(fsv_alignment, tile_size),
+            CeilDiv(in.X().v, tile_size) * in.Y().v,
+            in.Batch().v * CeilDiv(in.Feature().v, fsv_alignment) };
+        break;
+    case DataLayout::b_fs_zyx_fsv16:
+    case DataLayout::b_fs_zyx_fsv32:
+    case DataLayout::bs_fs_zyx_bsv16_fsv16:
+        gws = { CeilDiv(fsv_alignment, tile_size),
+            CeilDiv(in.X().v, tile_size) * in.Y().v * in.Z().v,
+            in.Batch().v * CeilDiv(in.Feature().v, fsv_alignment) };
+        break;
```

Comment:
I think in.Z().v will be 1 if the Z dimension is not used, so we can share the function for both cases..

Reply:
Also seems no need to take CeilDiv for fsv_alignment because it is always aligned to tile size..

Reply:
applied.